src/webgpu/shader/execution/memory_model/memory_model_setup.ts

```diff
@@ -191,47 +202,59 @@ export class MemoryModelTester {
     this.test = t;
     this.params = params;
 
-    testShader = shaderPreamble(accessValueType) + testShader;
-    resultShader = shaderPreamble(accessValueType) + resultShader;
+    const constants = `
+      const kNumBarriers = 1u;  // MAINTENANCE_TODO: make barrier not an array
+      const kMaxWorkgroups = ${params.maxWorkgroups}u;
+      const kScratchMemorySize = ${params.scratchMemorySize}u;
+      const kWorkgroupXSize = ${Math.min(256, t.device.limits.maxComputeWorkgroupSizeX)}u;
+    `;
+    testShader = shaderPreamble(accessValueType, constants) + testShader;
+    resultShader = shaderPreamble(accessValueType, constants) + resultShader;
 
     // set up buffers
     const testingThreads = this.params.workgroupSize * this.params.testingWorkgroups;
```

Comment:
Please use the same workgroup size throughout.  This now differs from the tweaked value set at line 209 for `kWorkgroupSize`.

Reply:
done?